src/model/GeneratorWindTurbine.cpp

```diff
@@ -0,0 +1,639 @@
+      bool result = setDouble(OS_Generator_WindTurbineFields::BladeChordArea, bladeChordArea);
+      return result;
+    }
+
+    void GeneratorWindTurbine_Impl::resetBladeChordArea() {
+      bool result = setString(OS_Generator_WindTurbineFields::BladeChordArea, "");
+      OS_ASSERT(result);
+    }
+
+    bool GeneratorWindTurbine_Impl::setBladeDragCoefficient(double bladeDragCoefficient) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::BladeDragCoefficient, bladeDragCoefficient);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setBladeLiftCoefficient(double bladeLiftCoefficient) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::BladeLiftCoefficient, bladeLiftCoefficient);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC1(double powerCoefficientC1) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC1, powerCoefficientC1);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC2(double powerCoefficientC2) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC2, powerCoefficientC2);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC3(double powerCoefficientC3) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC3, powerCoefficientC3);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC4(double powerCoefficientC4) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC4, powerCoefficientC4);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC5(double powerCoefficientC5) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC5, powerCoefficientC5);
+      return result;
+    }
+
+    bool GeneratorWindTurbine_Impl::setPowerCoefficientC6(double powerCoefficientC6) {
+      bool result = setDouble(OS_Generator_WindTurbineFields::PowerCoefficientC6, powerCoefficientC6);
+      return result;
+    }
+
+  }  // namespace detail
+
+  GeneratorWindTurbine::GeneratorWindTurbine(const Model& model) : Generator(GeneratorWindTurbine::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::GeneratorWindTurbine_Impl>());
+
+    setRotorType("HorizontalAxisWindTurbine");
+    setPowerControl("VariableSpeedVariablePitch");
+    setRatedRotorSpeed(41.0);  // from docs
+    setRotorDiameter(19.2);    // from docs
+    setOverallHeight(30.5);    // from docs
+    setNumberofBlades(3);
+    setRatedPower(55000.0);    // from docs
+    setRatedWindSpeed(11.0);   // from docs
+    setCutInWindSpeed(3.5);    // from docs
+    setCutOutWindSpeed(25.0);  // from docs
+    setFractionSystemEfficiency(0.835);
+    setMaximumTipSpeedRatio(5.0);
+    setMaximumPowerCoefficient(0.25);
+    setHeightforLocalAverageWindSpeed(50.0);
+    setBladeDragCoefficient(0.9);
+    setBladeLiftCoefficient(0.05);
```

Comment:
Since you default to `HorizontalAxisWindTurbine`, perhaps we should make all unrelated coefficients to 0, and add the Chord Area once it's made required?

```suggestion
    setBladeDragCoefficient(0.0);
    setBladeLiftCoefficient(0.0);
    setBladeChordArea(0.0);
```

Reply:
Actually, perhaps we should leave it as is...

Reply:
What should Blade Chord Area default to?

Reply:
It won't hurt to have it in the FT (and I wrote it before I realized you put sensible defaults here) so I pushed it anyways in ff5f76814e46a4d6b26723dd8638d1994b73033c

Reply:
@joseph-robertson I'll let you decide on this one, I'm fine with either zero or the current defaults you set!